src/mod/plugins/zoraxy_plugin/event.go

```diff
@@ -0,0 +1,120 @@
+package zoraxy_plugin
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// EventName represents the type of event
+type EventName string
+
+// EventPayload interface for all event payloads
+type EventPayload interface {
+	// GetName returns the event type
+	GetName() EventName
```

Comment:
I might wanna add a few more methods in the interface, but I guess we can merge this first

Reply:
Like `GetEvenSource()` to see who emitted the payload and `GetPayloadType()` for dumping the original data structure? These seems helpful for making IPC calls.

Reply:
@AnthonyMichaelTDM I would love to, but it is an interface not a struct. So if a plugin is developed for version `x` and we changed the interface for a bit, a ton of plugins will fail to build on version `x+1` . 

It would be better for us to figure out all the interfaces and try to minimize the change to this from a forward compatibility stand point. Or alternatively, I can label the v3.2.6 as a pre-release and try to figure out the rest of the interfaces before the next release. Which option would you prefer?

Reply:
That's a good point, how about we leave it be for now and implement the new functions in a new PR, which we finish before releasing 3.2.6?

Reply:
I'll take the emoji as approval, merge, and open an issue for adding the new functions